### PR TITLE
[Docs] Add DialoGPT

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR adds DialoGPT to the model page and links the models on the model page https://github.com/huggingface/transformers#model-architectures to the docs.

### CODE REVIEW

**Review by sshleifer — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM! Pinging @julien-c for the links to other repositories

File: docs/source/model_doc/dialo_gpt2.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+DialoGPT
```
Comment (julien-c) [inline docs/source/model_doc/dialo_gpt2.rst:1]: this file should be named dialogpt.rst, no?

File: docs/source/model_doc/dialo_gpt2.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Tips:
+
+- DialoGPT is a model with absolute position embeddings so it's usually advised to pad the inputs on
+  the right rather than the left.
+- DialoGPT was trained with a causal language modeling (CLM) objective on conversational data and is therefore powerful at response generation in open-domain dialogue systems.
+- DialoGPT enables the user to create a chat bot in just 10 lines of code as shown on `DialoGPT's model card <https://huggingface.co/microsoft/DialoGPT-medium>`_.
+ 
+
+DialoGPT's architecutre is based on the GPT2 model, so one can refer to GPT2's `docstring <https://huggingface.co/transformers/model_doc/gpt2.html>`_.
```
Comment (julien-c) [inline docs/source/model_doc/dialo_gpt2.rst:27]:
```suggestion
DialoGPT's architecture is based on the GPT2 model, so one can refer to GPT2's `docstring <https://huggingface.co/transformers/model_doc/gpt2.html>`_.
```

**Review by julien-c — APPROVED** [review]

File: docs/source/model_doc/dialo_gpt2.rst (hunk 1 above)
Comment (julien-c) [reply docs/source/model_doc/dialo_gpt2.rst:1]: (i.e. why the 2?)

Comment (patrickvonplaten) [reply docs/source/model_doc/dialo_gpt2.rst:1]: true!

**Pull request merged**